File: lib/models/notification.dart
class Notification_Model {
  final String id;
  final String userId;
  final String sellerId;
  final String type;
  final String productId;
  final String message;
  final bool isRead;
  final DateTime createdAt;

  Notification_Model({
    required this.id,
    required this.userId,
    required this.sellerId,
    required this.type,
    required this.productId,
    required this.message,
    required this.isRead,
    required this.createdAt,
  });

  factory Notification_Model.fromMap(Map<String, dynamic> map) {
    return Notification_Model(
      id: map['_id'] ?? '',
      userId: map['userId'] is Map ? map['userId']['_id'].toString() : (map['userId'] ?? ''),
      sellerId: map['sellerId'] is Map ? map['sellerId']['_id'].toString() : (map['sellerId'] ?? ''),
      type: map['type'] ?? '',
      productId: map['productId'] is Map ? map['productId']['_id'].toString() : (map['productId'] ?? ''),
      message: map['message'] ?? '',
      isRead: map['isRead'] ?? false,
      createdAt: DateTime.parse(map['createdAt']),
    );
  }
}

File: server/models/notification.js
const mongoose = require("mongoose");

const notificationSchema = mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    sellerId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    type: {
        type: String,
        enum: ['new_product', 'update_product', 'discount'],
        required: true
    },
    productId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Product',
        required: true
    },
    message: {
        type: String,
        required: true
    },
    isRead: {
        type: Boolean,
        default: false
    },
    createdAt: {
        type: Date,
        default: Date.now
    }
});

const Notification = mongoose.model("Notification", notificationSchema);
module.exports = Notification;